File: frontend/src/app/shared/model/vo/filter/vacation-filter-vo.ts
import { AuditableFilterVO } from './auditable-filter-vo';
import { EmployeeFilterVO } from './employee-filter-vo';

export interface VacationFilterVO extends AuditableFilterVO {
    //Data inicial da férias.
	startDate: Date[];

	//Data final da férias.
	endDate: Date[];

	//Funcionário do qual pertence a férias.
	employee: EmployeeFilterVO;
}

File: backend/src/main/java/br/com/cams7/cadferias/model/vo/filter/VacationFilterVO.java
/**
 * 
 */
package br.com.cams7.cadferias.model.vo.filter;

import java.time.LocalDate;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author ceanm
 *
 */
@ApiModel(description = "Filtro de busca com os dados da férias.")
@Getter
@Setter
@NoArgsConstructor
@ToString
public class VacationFilterVO extends AuditableFilterVO {

	@ApiModelProperty(notes = "Data inicial da férias.", required = false, position = 4)
	private LocalDate[] startDate;

	@ApiModelProperty(notes = "Data final da férias.", required = false, position = 5)
	private LocalDate[] endDate;

	@ApiModelProperty(notes = "Funcionário do qual pertence a férias.", required = false, position = 6)
	private EmployeeFilterVO employee;
}
